lms/djangoapps/commerce/utils.py

```diff
@@ -109,3 +117,247 @@ def upgrade_url(self, user, course_key):
+            # result in opening a refund request.
+            msg = 'Skipping refund support notification for non-verified mode for user [%s], course [%s], mode: [%s]'
+            course_identifier = course_product.course_id
+            if course_product.uuid:
+                course_identifier = str(course_product.uuid)
+                msg = ('Skipping refund support notification for non-verified mode for user [%s], '
+                       'course entitlement [%s], mode: [%s]')
+            log.info(
+                msg,
+                course_product.user.id,
+                course_identifier,
+                course_product.mode,
+            )
+            return False
+        else:
+            try:
+                _send_refund_notification(course_product, refunds_requiring_approval)
+            except:  # pylint: disable=bare-except
+                # Unable to send notification to Support, do not break as this method is used by Signals
+                log.warning('Could not send support notification for refund.', exc_info=True)
+                return False
+    return True
+
+
+def _send_refund_notification(course_product, refund_ids):
+    """ Notify the support team of the refund request. """
+
+    tags = ['auto_refund']
+
+    if theming_helpers.is_request_in_themed_site():
+        # this is not presently supported with the external service.
+        raise NotImplementedError("Unable to send refund processing emails to support teams.")
+
+    student = course_product.user
+    subject = _("[Refund] User-Requested Refund")
+    body = _generate_refund_notification_body(student, refund_ids)
+    requester_name = student.profile.name or student.username
+    create_zendesk_ticket(requester_name, student.email, subject, body, tags)
+
+
+def _generate_refund_notification_body(student, refund_ids):  # pylint: disable=invalid-name
+    """ Returns a refund notification message body. """
+    msg = _(
+        'A refund request has been initiated for {username} ({email}). '
+        'To process this request, please visit the link(s) below.'
+    ).format(username=student.username, email=student.email)
+
+    ecommerce_url_root = configuration_helpers.get_value(
+        'ECOMMERCE_PUBLIC_URL_ROOT', settings.ECOMMERCE_PUBLIC_URL_ROOT,
+    )
+    refund_urls = [urljoin(ecommerce_url_root, '/dashboard/refunds/{}/'.format(refund_id))
+                   for refund_id in refund_ids]
+
+    # emails contained in this message could contain unicode characters so encode as such
+    return u'{msg}\n\n{urls}'.format(msg=msg, urls='\n'.join(refund_urls))
+
+
+def create_zendesk_ticket(requester_name, requester_email, subject, body, tags=None):
+    """ Create a Zendesk ticket via API. """
+    if not (settings.ZENDESK_URL and settings.ZENDESK_USER and settings.ZENDESK_API_KEY):
+        log.debug('Zendesk is not configured. Cannot create a ticket.')
+        return
```

Comment:
I know you just moved this code, but maybe this should return false so that we don't forget that these refunds were requested, especially if we've already gone ahead and removed their enrollment.

Also, this probably should be logged higher than debug.

Reply:
Take a look at the changes I made here.